File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 2
  def self.compute(left, right)
    raise ArgumentError unless left.length == right.length
    left.chars.zip(right.chars).count do |lchar, rchar|
      lchar != rchar
    end
  end
end

Hamming.compute("AAA", "AAA")

File: java/hamming/src/main/java/Hamming.java
public class Hamming {
    public static Integer compute(String left, String right) {
        if (left.length() != right.length())
            throw new IllegalArgumentException("Strands must be same length");
        Integer dist = 0;
        for (int i = 0; i < left.length(); i++) {
            if (left.charAt(i) != right.charAt(i))
                dist++;
        }
        return dist;
    }

}

